src/interface/chat/user-input.tsx

```diff
@@ -0,0 +1,28 @@
+import React, { useCallback } from 'react';
+import { Box, Text } from 'ink';
+import { TextInput } from '@inkjs/ui';
+
+type UserInputProps = {
+  onSubmit: (value: string) => void;
+  visible: boolean;
+};
+
+export const UserInput = ({ onSubmit, visible }: UserInputProps) => {
+  const handleInput = useCallback(
```

Comment:
There seems to be little point in wrapping this in `useCallback` This control does not get rerendered on typing, as there is not capturing current value during typing.

Reply:
Done